File: examples/sdk-query.py
"""End-to-end example for the Coderix Python SDK (`coderix-sdk`).

Run after installing the SDK (``pip install -e packages/coderix-sdk-python``):

    python examples/sdk-query.py

Mirrors claude-code-sdk's Python ``query()`` + ``CoderixSDKClient``.
"""

import asyncio

from coderix_sdk import query, CoderixSDKClient


async def one_shot() -> None:
    print("=== query() ===")
    async for msg in query(
        prompt="Reply with exactly: hello",
        options={"permission_mode": "plan", "max_turns": 3},
    ):
        t = msg.get("type")
        if t == "system":
            print(f"[system] {msg.get('subtype')} session={msg.get('session_id')}")
        elif t == "result":
            print(
                f"[result] {msg.get('subtype')} "
                f"turns={msg.get('num_turns')} cost=${msg.get('total_cost_usd', 0):.4f}"
            )
            print(f"         {msg.get('result')}")
        elif t == "assistant":
            content = msg.get("message", {}).get("content")
            if isinstance(content, str):
                print(f"[assistant] {content}")


async def client() -> None:
    print("\n=== CoderixSDKClient ===")
    c = CoderixSDKClient(options={"permission_mode": "plan"})
    await c.connect()
    try:
        async for msg in c.query(prompt="Say hi in one word"):
            if msg.get("type") == "result":
                print(f"[result] {msg.get('subtype')}: {msg.get('result')}")
    finally:
        await c.disconnect()


async def main() -> None:
    await one_shot()
    await client()


if __name__ == "__main__":
    asyncio.run(main())

File: examples/sdk-query.ts
/**
 * End-to-end example for the Coderix TypeScript SDK (`@coderix/sdk`).
 *
 * Run from the repo root (after `pnpm install` + `pnpm build`):
 *
 *   npx tsx examples/sdk-query.ts
 *
 * Mirrors claude-code-sdk's `query()` async generator.
 */

import { query, CoderixSDKClient } from '@coderix/sdk';

async function oneShot() {
  console.log('=== query() ===');
  for await (const msg of query({
    prompt: 'Reply with exactly: hello',
    options: { permissionMode: 'plan', maxTurns: 3, includePartialMessages: false },
  })) {
    switch (msg.type) {
      case 'system':
        console.log(`[system] ${msg.subtype} session=${msg.session_id}`);
        break;
      case 'assistant': {
        const content = typeof msg.message.content === 'string'
          ? msg.message.content
          : msg.message.content.filter((b) => b.type === 'text').map((b) => b.text ?? '').join('');
        console.log(`[assistant] ${content}`);
        break;
      }
      case 'result':
        console.log(`[result] ${msg.subtype} turns=${msg.num_turns} cost=$${msg.total_cost_usd.toFixed(4)}`);
        console.log(`         ${msg.result}`);
        break;
      default:
        break;
    }
  }
}

async function client() {
  console.log('\n=== CoderixSDKClient ===');
  const c = new CoderixSDKClient({ permissionMode: 'plan' });
  await c.connect();
  try {
    for await (const msg of c.query({ prompt: 'Say hi in one word' })) {
      if (msg.type === 'result') {
        console.log(`[result] ${msg.subtype}: ${msg.result}`);
      }
    }
  } finally {
    await c.disconnect();
  }
}

async function main() {
  await oneShot();
  await client();
}

main().catch((err) => {
  console.error('Example failed:', err);
  process.exit(1);
});
